File: expense.js
const splitType = document.getElementById("split-type");
function toggleFields() {
    if (splitType.value === "friend") {
        friendOwes.style.display = "block";
        friendName.style.display = "block";
        groupFields.style.display = "none";
        groupCost.style.display = "none";
        groupName.style.display = "none";

    } else if (splitType.value === "group") {
        friendOwes.style.display = "none";
        friendName.style.display = "none";
        groupFields.style.display = "block";
        groupCost.style.display = "block";
        groupName.style.display = "block";
    }
}

const friendOwes = document.getElementById("friend-owes");
const friendName = document.getElementById("friend-name");
const groupFields = document.getElementById("group-fields");
const groupCost = document.getElementById("group-cost");
const groupMem = document.getElementById("group-size");
const groupName = document.getElementById("group-name");

const totAmount = document.getElementById("amount");
const userAmount = document.getElementById("amount_u");


splitType.addEventListener("change", toggleFields);
totAmount.addEventListener("input", Calculateamount);
userAmount.addEventListener("input", Calculateamount);

function Calculateamount() {
    const tot_amount = totAmount.value;
    const user_amount = userAmount.value;
    const owed_amount = user_amount - (tot_amount) / 2;
    var amount_owed = document.getElementById("friend-cost-per-person");
    amount_owed.innerHTML = `<p>${owed_amount}</p>`;
}

splitType.addEventListener("change", toggleFields);
totAmount.addEventListener("input", CalculateGroupAmount);
userAmount.addEventListener("input", CalculateGroupAmount);


function CalculateGroupAmount() {
    const tot_amount = totAmount.value;
    const user_amount = userAmount.value;
    const members = groupMem.value;
    const owed_amount2 = ((user_amount - (tot_amount) / members) / members);
    var amount_owed = document.getElementById("group-cost-per-person");
    amount_owed.innerHTML = `<p>${owed_amount2}</p>`;
}

File: expense.py
import sqlite3
from flask import Flask, render_template, request,jsonify,redirect,url_for,flash,get_flashed_messages

app=Flask(__name__)

conn = sqlite3.connect('database.db')
c = conn.cursor()


@app.route('/addBill', methods=['GET','POST'])
def calculate_expense():
    split_type = request.form['split-type']
   
    dic=[]
    d=c.execute("SELECT Name FROM login_log")
    for x in d:
        for s in x:
            dic.append(s)
    user_name = dic[-1]

    if split_type == 'Friend':
        friend_name = request.form['friend_name']
        owed_amount = request.form['friend-cost-per-person']
        amount_you_owe_query = "SELECT owebyU FROM Info WHERE username=? AND frname = ?"
        amount_you_are_owed_query = "SELECT owebyfr FROM Info WHERE username=? AND frname = ?"
        c.execute(amount_you_owe_query, (friend_name,))
        amount_you_owe = c.fetchone()[0]
        c.execute(amount_you_are_owed_query, (friend_name,))
        amount_you_are_owed = c.fetchone()[0]
        if owed_amount >= 0:
            amount_you_owe = amount_you_owe + owed_amount
            c.execute("UPDATE Info SET owebyU = ? WHERE username=? AND frname = ?", (amount_you_owe, user_name,friend_name,))
            c.execute("UPDATE Info SET owebyfr = ? WHERE username=? AND frname = ?", (amount_you_owe, friend_name, user_name))
        elif owed_amount < 0:
            amount_you_are_owed = amount_you_are_owed - owed_amount
            c.execute("UPDATE Info SET owebyfr = ? WHERE username=? AND frname = ?", (amount_you_are_owed, user_name, friend_name))
            c.execute("UPDATE Info SET owebyU = ? WHERE username=? AND frname = ?", (amount_you_are_owed, friend_name, user_name))



    elif split_type=="Group":
        group_name = request.form['group_name']
        c.execute("SELECT username FROM Groups WHERE Gname=?", (group_name,))
        friend_names = [row[0] for row in c.fetchall()]

# Add labels for each friend to the form
        for friend_name in friend_names:
            friend_label = f"<label for='{friend_name}'>{friend_name} Paid</label>"
            friend_input = f"<input type='number' id='{friend_name}' name='{friend_name}'>"
            friend_fields = f"<div class='form-group-split-fields split-friend'>{friend_label}{friend_input}</div>"
    # Append the friend fields to the form
        form_element = driver.find_element_by_tag_name('form')
        form_element.insert_child(-2, friend_fields)


    



    conn.commit()
    conn.close()

if __name__ == '__main__':
    app.run(debug=True)
